topics/admin/tutorials/job-destinations/tutorial.md

```diff
@@ -316,107 +354,70 @@ Dynamic destinations allow you to write custom python code to dispatch jobs base
->         dest: "{{ galaxy_config.galaxy.dependency_resolvers_config_file }}"
->    +  - src: templates/galaxy/config/tool_destinations.yml
->    +    dest: "{{ galaxy_config.galaxy.tool_destinations_config_file }}"
->     
->     galaxy_local_tools:
->     - testing.xml
+>    @@ -40,6 +40,7 @@ galaxy_config:
+>                 function: map_tool_to_destination
+>                 rules_module: tpv.rules
+>                 tpv_config_files:
+>    +              - https://raw.githubusercontent.com/galaxyproject/tpv-shared-database/main/tools.yml
+>                   - config/tpv_rules_local.yml
+>           tools:
+>             - class: local # these special tools that aren't parameterized for remote execution - expression tools, upload, etc
 >    {% endraw %}
 >    ```
->    {: data-commit="Deploy tool destinations config file"}
+>    {: data-commit="Importing TPV shared database via job conf"}
 >
-> 3. We need to update Galaxy's job configuration to use this rule. Open `templates/galaxy/config/job_conf.yml.j2` and add a DTD destination.
->    Also, comment out or remove the previous `<tool>` definition for the `testing` tool, and replace it with a mapping to the dtd destination like so:
+> Note how TPV allows the file to be imported directly via its http url. As many local and remote rule files as necessary can be combined, with rule files specified later overriding
+> any previously specified rule files. The TPV shared database does not define destinations, only cores and mem settings, as well as any required environment vars.
+> Take a look at the shared database of rules and note that some tools have very large recommended memory settings, which may or may not be available within your local cluster.
+> Nevertheless, you may still wish to execute these tools with memory adjusted to suit your cluster's capabilities. 
+>
+> 2. Edit your `files/galaxy/config/tpv_rules_local.yml` and make the following changes.
 >
 >    {% raw %}
 >    ```diff
->    --- a/group_vars/galaxyservers.yml
->    +++ b/group_vars/galaxyservers.yml
->    @@ -66,11 +66,14 @@ galaxy_job_config:
->           dynamic_admin_only:
->             runner: dynamic
->             function: admin_only
->    +      dtd:
->    +        runner: dynamic
->    +        type: dtd
->       tools:
->         - class: local # these special tools that aren't parameterized for remote execution - expression tools, upload, etc
->           environment: local_env
->         - id: testing
->    -      environment: dynamic_admin_only
->    +      environment: dtd
->     
->     galaxy_config:
->       galaxy:
+>    --- a/files/galaxy/config/tpv_rules_local.yml
+>    +++ b/files/galaxy/config/tpv_rules_local.yml
+>    @@ -30,7 +30,10 @@ destinations:
+>       slurm:
+>         inherits: singularity
+>         runner: slurm
+>    -    max_accepted_cores: 16
+>    +    max_accepted_cores: 24
+>    +    max_accepted_mem: 256
+>    +    max_cores: 16
+>    +    max_mem: 128
```

Comment:
note that training VMs have like... 2 core and 8gb ram max usually. Should we decrease these numbers (assuming they're in GB?) (or is it MB?) to match the training VMs then, and then provide additional examples of "production" numbers in a tip box or so as "generally a good set of numbers"?

Reply:
Sounds like a good idea to reduce to 2 and 8.